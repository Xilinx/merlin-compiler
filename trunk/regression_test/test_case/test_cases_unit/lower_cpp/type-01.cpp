//(C) Copyright 2016-2021 Xilinx, Inc.
//All Rights Reserved.
//
//Licensed to the Apache Software Foundation (ASF) under one
//or more contributor license agreements.  See the NOTICE file
//distributed with this work for additional information
//regarding copyright ownership.  The ASF licenses this file
//to you under the Apache License, Version 2.0 (the
//"License"); you may not use this file except in compliance
//with the License.  You may obtain a copy of the License at
//
//  http://www.apache.org/licenses/LICENSE-2.0
//
//Unless required by applicable law or agreed to in writing,
//software distributed under the License is distributed on an
//"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
//KIND, either express or implied.  See the License for the
//specific language governing permissions and limitations
//under the License. (edited)
namespace st {
  enum my_em {ONE};
  struct A {};
  typedef int integer;
};

#pragma ACCEL kernel
void k()
{
  int f = 1;
  enum st::my_em a = (enum st::my_em)f;
  int s = sizeof(struct st::A);
  int s1 = sizeof(struct st::A **);
  int s2 = sizeof(struct st::A *);
  int s3 = sizeof(struct st::A [2]);
  int s4 = sizeof(const struct st::A [2][3]);
  int t = sizeof(st::integer);
  int t2 = sizeof(st::integer [2]);
  int t3 = sizeof(st::integer *);
  const struct st::A arr[2];
  struct st::A *p = (struct st::A *)arr;
}
