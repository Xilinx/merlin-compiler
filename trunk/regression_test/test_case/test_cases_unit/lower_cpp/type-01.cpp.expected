
struct ::st_A 
{
}
;
typedef int st_integer;
enum _st_my_em {MERLIN_MAX_ENUM_0=1L} ;
namespace st
{
static void __merlin_dummy_SgEnumDeclaration_enum_my_em_();
static void __merlin_dummy_SgClassDeclaration_struct_A_();
static void __merlin_dummy_SgTypedefDeclaration_typedef_in_();
}
#pragma ACCEL kernel

void k()
{
int f = 1;
enum _st_my_em a = (enum _st_my_em )f;
int s = (sizeof(struct st_A ));
int s1 = (sizeof(struct st_A **));
int s2 = (sizeof(struct st_A *));
int s3 = (sizeof(struct st_A [2]));
int s4 = (sizeof(const struct st_A [2][3]));
int t = (sizeof(::st_integer ));
int t2 = (sizeof(::st_integer [2]));
int t3 = (sizeof(::st_integer *));
const struct st_A arr[2];
struct st_A *p = (struct st_A *)arr;
}
